### ISSUE TITLE
Split client out of main __init__.py file

### ISSUE THREAD

**Issue by rlotun** [joestump/python-oauth2#39]
The current structure of the library mixes OAuth signing tools with http client library code - specifically _synchronous_ http library code. We use Twisted, and would like to use all the great signing code without having to have httplib2 installed. 

I'm thinking it might be nicer to have it moved to the clients sub-package and possibly have some import logic in **init**.py with a try/except ImportError around httplib2.

**Comment by jaitaiwan** [comment]
Fork no longer exists but a valid request none-the-less. Thoughts @joestump?

**Comment by joestump** [comment]
I agree we should split up the monolith that is `__init__.py`. Was just thinking about that this morning.

**Comment by jaitaiwan** [comment]
Easy enough for 1.9RC1 or schedule for 2.0?

**Comment by joestump** [comment]
I think we should get 1.9RC1 out with as many bug fixes so my vote is to save this work for 2.0.
